db/basic/db.go

```diff
@@ -209,6 +209,32 @@ func (db *basicDB) Insert(ctx context.Context, table string, key string, values
 	return nil
 }
 
+func (db *basicDB) BatchInsert(ctx context.Context, table string, keys []string, values []map[string][]byte) error {
+	state := ctx.Value(stateKey).(*basicState)
+
+	db.delay(ctx, state)
+
+	if !db.verbose {
+		return nil
+	}
+	buf := state.buf
+	for i, key := range keys {
+		s := fmt.Sprintf("INSERT %s %s [ ", table, key)
```

Comment:
maybe you can extract common part from Insert and BatchInsert

Reply:
+1